### DISCUSSION TITLE
Conditioning on group with additional covariance matrix

### DISCUSSION THREAD

**Discussion by scutisorex** [lindeloev/mcp#167] [Q&A]
Hi all, 

I'm trying to use mcp to execute a phylogenetically-informed multiple change point regression of the relationship between animal body mass and other morphological variables. 

In brms, the syntax for conditioning on a phylogenetic tree (to illustrate the likelihood of similarity due to relatedness in the animals in the analysis) is as follows, from the vignette by Paul Bürkner: https://cran.r-project.org/web/packages/brms/vignettes/brms_phylogenetics.html:

model_simple <- brm(
  phen ~ cofactor + (1|gr(phylo, cov = A)),
  data = data_simple,
  family = gaussian(),
  data2 = list(A = A),
  prior = c(
    prior(normal(0, 10), "b"),
    prior(normal(0, 50), "Intercept"),
    prior(student_t(3, 0, 20), "sd"),
    prior(student_t(3, 0, 20), "sigma")
  )
)

So in this example, brms allows you to specify a covariance matrix that is made directly from your phylogenetic tree (here, cov = A). With mcp, I have been able to condition on phylo alone, as it is a column in my dataset, specified like this: 

model = list(
  response ~ 0 + predictor+(1|phylo),
  ~ 0 + predictor+(1|phylo), 
  ~ 0 + predictor+(1|phylo)
)

However, phylo is a character that specifies group rather than a quantitative representation of the covariance among my data points. To be able to use the covariance matrix, I need to be able to specify something like the "data2" term in the brms code. as far as I have found so far, mcp doesn't support this. Additionally, I have no idea if the 1|gr(phylo, cov = A) syntax would work within mcp. 

Long story short, I'd like to be able to condition on a group with a separate covariance matrix, which I would like to be able to specify in the mcp call. Does anyone have ideas, or workarounds, or any clue how difficult this would be to implement?

Many thanks, 

Steph

**Comment by lindeloev** [comment]
Thanks for the detailed request, @scutisorex! Specifying the covariance structure on the RHS is a frequent request. There's no simple workaround right now. You *could* do this, if you're open to do a bit of work:

1. Build a model using `mcp` that's as close as you can get.
2. Look at the JAGS code in fit$jags_code and adapt it. https://github.com/andrewcparnell/jags_examples is a great repo to see relevant examples, e.g., random effects etc.
3. Run everything yourself using the `rjags`-package or the like.

Best-case, it'll be 6 months until I get to work on just basic RHS-covariance, just to set expectations. Time is limited these days.
